3_train/src/train.py

```diff
@@ -0,0 +1,465 @@
+
+    :param train_ds: Dataset for training data
+    :param valid_ds: Dataset for validation data
+    :param batch_size: Number of elements in each training batch
+    :returns: Tuple of training data loader and validation data loader
+
+    """
+    return (
+        DataLoader(train_ds, batch_size=batch_size, shuffle=True),
+        # The validate set is not used for training with backprop, so we can
+        # afford its batch size to be larger and it doesn't need to be shuffled
+        DataLoader(valid_ds, batch_size=batch_size * 2),
+    )
+
+
+def get_data(npz_filepath,
+             dynamic_features_use,
+             static_features_use,
+             depths_use,
+             batch_size,
+             valid_frac,
+             seed):
+    """
+    Get train and validate dataloaders from npz file
+    
+    Patterned after https://pytorch.org/tutorials/beginner/nn_tutorial.html#create-fit-and-get-data
+
+    :param npz_filepath: Name and path to .npz data file
+    :param dynamic_features_use: List of dynamic features to include
+    :param static_features_use: List of static features to include
+    :param depths_use: List of depth values to include
+    :param batch_size: Number of elements in each training batch
+    :param valid_frac: The fraction of data samples to put into the validation set
+    :param seed: Integer seed for random number generator
+
+    """
+
+    # Check that valid_frac is a fraction
+    if (valid_frac < 0) or (valid_frac > 1):
+        raise ValueError("valid_frac must be between 0 and 1")
+
+    # Training data
+    data_npz = np.load(npz_filepath)
+
+    # Determine which depths and features to use
+    # Input features are drivers, clarity, ice flags, and static attributes, in that order.
+    # Combine full set into one list
+    elements_all = (data_npz['depths_all'].tolist() +
+                    data_npz['dynamic_features_all'].tolist() +
+                    data_npz['static_features_all'].tolist())
+    # Combine set to use into one list
+    elements_use = (depths_use + dynamic_features_use + static_features_use)
+    # Select set to use out of full set
+    idx_to_use = [elements_all.index(e) for e in elements_use]
+    # Shape of data is (# sequences, sequence_length, # depths + # input features)
+    data_array_to_use = data_npz['data'][:,:,idx_to_use]
+
+    n_depths = len(depths_use)
+    n_dynamic = len(dynamic_features_use)
+    n_static = len(static_features_use)
+
+    # Train/validate split
+    dataset = SequenceDataset(
+        data_array_to_use,
+        n_depths,
+        n_dynamic,
+        n_static
+    )
+    train_subset, valid_subset = split(dataset, 1-valid_frac, seed)
+    return get_data_loaders(train_subset, valid_subset, batch_size)
+
+
+def get_model(
+    n_depths,
+    n_dynamic,
+    n_static,
+    hidden_size,
+    initial_forget_bias,
+    dropout,
+    concat_static
+):
+    """
+    Create LSTM torch model
```

Comment:
Might be good to have the documentation mention that this is LSTM _and_ EA LSTM compatible (because that Model class can be both)

Reply:
Yes - added! I clarified that in a couple of other comments, too.